### ISSUE TITLE
AutoTFT can't be trained on multiple GPUs

### ISSUE THREAD

**Issue by jmoralez** [Nixtla/neuralforecast#1110]
### What happened + What you expected to happen

When using DDP (the default strategy in pytorch lightning for multiple GPUs) the training fails right after starting. It seems to be related to some parameters not producing gradients. By setting some env variables to get more information as suggested in https://github.com/Lightning-AI/pytorch-lightning/issues/17212#issuecomment-1632222390 I see the following:

```
[rank1]:[I816 21:41:23.034545571 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.3.glu.lin.bias did not get gradient i
n backwards pass.                                                                                                                       
[rank1]:[I816 21:41:23.034555851 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.3.glu.lin.weight did not get gradient
 in backwards pass.                                                                                                                     
[rank1]:[I816 21:41:23.034559131 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.3.lin_i.bias did not get gradient in 
backwards pass.                                                                                                                         
[rank1]:[I816 21:41:23.034562242 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.3.lin_i.weight did not get gradient i
n backwards pass.                                                                                                                       
[rank1]:[I816 21:41:23.034565272 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.3.lin_a.bias did not get gradient in 
backwards pass.                                                                                                                         
[rank1]:[I816 21:41:23.034573542 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.3.lin_a.weight did not get gradient i
n backwards pass.
[rank1]:[I816 21:41:23.034578032 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.3.layer_norm.ln.bias did not get grad
ient in backwards pass.
[rank1]:[I816 21:41:23.034580892 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.3.layer_norm.ln.weight did not get gr
adient in backwards pass.
[rank1]:[I816 21:41:23.034584372 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.2.glu.lin.bias did not get gradient i
n backwards pass.
[rank1]:[I816 21:41:23.034588952 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.2.glu.lin.weight did not get gradient
 in backwards pass.
[rank1]:[I816 21:41:23.034592102 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.2.lin_i.bias did not get gradient in 
backwards pass.
[rank1]:[I816 21:41:23.034596452 reducer.cpp:1989] [Rank 1] Parameter: static_encoder.context_grns.2.lin_i.weight did not get gradient i
n backwards pass.
```

So probably the following
https://github.com/Nixtla/neuralforecast/blob/6aed50ecf8316c6325c1677362e89e0138325fe1/neuralforecast/models/tft.py#L513-L517
should only be added if `self.stat_exog_size > 0`

### Versions / Dependencies

1.7.4

### Reproduction script

```python
import optuna
from neuralforecast import NeuralForecast
from neuralforecast.auto import AutoTFT
from utilsforecast.data import generate_series


def config(trial):
    return {'max_steps': 5, 'input_size': 10, 'barebones': True}

series = generate_series(5)
models = [
    AutoTFT(
        h=10,
        config=config,
        num_samples=2,
        backend='optuna',
        search_alg=optuna.samplers.TPESampler(seed=0)
    )
]
nf = NeuralForecast(models=models, freq='D')
nf.fit(series)
```

### Issue Severity

High: It blocks me from completing my task.
